ui/src/features/Dashboard/components/Nav/components/WorkspaceNavigation.tsx

```diff
@@ -9,20 +9,20 @@ import {
   DropdownMenuItem,
   DropdownMenuTrigger,
 } from "@flow/components";
+import { useGetWorkspace } from "@flow/lib/gql";
 import { cn } from "@flow/lib/utils";
-import { workspaces } from "@flow/mock_data/workspaceData";
-import { useCurrentProject, useCurrentWorkspace } from "@flow/stores";
+import { useCurrentWorkspace } from "@flow/stores";
 import { Workspace } from "@flow/types";
 
 const WorkspaceNavigation: React.FC = () => {
   const [currentWorkspace, setCurrentWorkspace] = useCurrentWorkspace();
-  const [, setCurrentProject] = useCurrentProject();
-  const navigate = useNavigate({ from: "/workspace/$workspaceId" });
+  const { workspaces } = useGetWorkspace();
+  const navigate = useNavigate();
 
-  const handleWorkspaceChange = (workspace: Workspace) => {
-    setCurrentProject(undefined);
+  const handleWorkspaceChange = async (workspace: Workspace) => {
+    const route = window.location.pathname;
```

Comment:
react router has a useParams hook that you can use to get route information.

Reply:
I need the full route and I have the workspace ID from the currentWorkspace state. 
This is not correct in 100% of the places, it'll fail in some cases like when the 
/workspacesId/project/projectId is open 
For a proper implementation  I want to add a check in the routes/[route].tsx as well.